clang-tools-extra/test/clang-tidy/checkers/bugprone/capturing-this-by-field.cpp

```diff
@@ -0,0 +1,144 @@
+// RUN: %check_clang_tidy -std=c++11-or-later %s bugprone-capturing-this-by-field %t -- -config="{CheckOptions: {bugprone-capturing-this-by-field.FunctionWrapperTypes: '::std::function;::Fn'}}" --
+
+namespace std {
+
+template<class Fn>
+class function;
+
+template<class R, class ...Args>
+class function<R(Args...)> {
+public:
+  function() noexcept;
+  template<class F> function(F &&);
+};
+
+} // namespace std
+
+struct Fn {
+  template<class F> Fn(F &&);
+};
+
+struct Basic {
+  Basic() : Captured([this]() { static_cast<void>(this); }) {}
```

Comment:
`[&]` and `[=]` also captures "this", not only `[this]`

Reply:
add test